File: src/BusinessLogic/Handlers/MessageHandlers/KnownInlineAnswerMessageHandler.cs
using Bounan.Bot.BusinessLogic.Constants;
using Bounan.Bot.TelegramBot.Telegram;
using Microsoft.Extensions.Logging;
using Telegram.Bot;
using Telegram.Bot.Types;

namespace Bounan.Bot.BusinessLogic.Handlers.MessageHandlers;

public class KnownInlineAnswerMessageHandler(
    ILogger<KnownInlineAnswerMessageHandler> logger,
    ITelegramBotClient botClient) : IMessageHandler
{
    public static bool CanHandle(Message item) => item.Text is
        KnownInlineAnswers.AnimeUnavailable or KnownInlineAnswers.NoResults or KnownInlineAnswers.NoRelatedAnime;

    public Task HandleAsync(Message updateItem, CancellationToken cancellationToken)
    {
        ArgumentNullException.ThrowIfNull(updateItem);
        ArgumentNullException.ThrowIfNull(updateItem.Text);

        logger.LogInformation("Handling known inline answer {Text}", updateItem.Text);

        return botClient.DeleteMessageAsync(
            updateItem.Chat.Id,
            updateItem.MessageId,
            cancellationToken: cancellationToken);
    }
}

File: src/handlers/on-webhook/event-handlers/message-handlers/known-inline-answer-message-handler.ts
import { MessageHandler } from '../query-handler';
import { Message } from 'telegram-bot-api-lightweight-client/src/types';
import { deleteMessage } from 'telegram-bot-api-lightweight-client/src/client';
import { assert } from '../../../../shared/helpers/assert';
import { KnownInlineAnswers } from '../../constants/known-inline-answers';

const canHandle = (message: Message): boolean =>
    Object.values(KnownInlineAnswers).includes(message.text as KnownInlineAnswers);

const handler: MessageHandler = async (message) => {
    assert(!!message?.chat?.id);
    assert(!!message?.text);

    console.log('Handling known inline answer: ', message.text);

    const result = await deleteMessage({
        chat_id: message.chat.id,
        message_id: message.message_id,
    });
    assert(result.ok, () => JSON.stringify(result));
};

export const knownInlineAnswerMessageHandler = {
    canHandle,
    handler,
};